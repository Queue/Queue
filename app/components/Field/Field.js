//
// Field

import React, { Component } from 'react';
import { View, TextInput, Text, StyleSheet } from 'react-native';

import Colors from '../../lib/colors';

export default class Field extends Component {
  constructor(props) {
    super(props);
  }

  hasLabel() {
    if (this.props.label) {
      return (
        <Text style={styles.label}>{this.props.label}</Text>
      );
    }
  }

  render() {
    // text type
    if (this.props.type === 'text') {
      return (
        <View style={styles.fieldContainer}>
          {this.hasLabel()}
          <TextInput
            style={styles.fieldBase}
            {...this.props}
          />
        </View>
      );
    // if no type is specified
    } else {
      return (
        <Text style={{color: Colors.error}}>
          No Input Type
        </Text>
      );
    }
  }
}

const styles = StyleSheet.create({
  label: {
    letterSpacing: 1,
    marginBottom: 5,
    color: 'grey'
  },
  fieldContainer: {
    marginTop: 10,
    marginBottom: 10
  },
  fieldBase: {
    paddingLeft: 15,
    height: 45,
    borderColor: Colors.primaryBackground,
    borderWidth: 1
  },
  textareaInput: {
    paddingTop: 7,
    paddingLeft: 15,
    fontSize: 17,
    height: 140,
    borderColor: Colors.primaryBackground,
    borderWidth: 1
  }
});
