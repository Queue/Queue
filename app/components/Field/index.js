//
// Field index.js

import Field from './Field';
export default Field;
