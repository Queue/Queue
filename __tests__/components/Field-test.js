//
// Field component test

import React from 'react';
import Field from '../../app/components/Field';
import renderer from 'react-test-renderer';

let testFunc = () => {
  console.warn('');
};

it('renders correctly', () => {
  const tree = renderer.create(
    <Field
      onPress={testFunc}
      label={'Log In'}
      type={'text'}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
