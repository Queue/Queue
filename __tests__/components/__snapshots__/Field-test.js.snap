exports[`test renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "grey",
        "letterSpacing": 1,
        "marginBottom": 5,
      }
    }>
    Log In
  </Text>
  <TextInput
    label="Log In"
    onPress={[Function]}
    style={
      Object {
        "borderColor": "#eaeaea",
        "borderWidth": 1,
        "height": 45,
        "paddingLeft": 15,
      }
    }
    type="text" />
</View>
`;
